### ISSUE TITLE
Bash completion errors on macos which do not occur in standard terminal

### ISSUE THREAD

**Issue by imsodin** [kovidgoyal/kitty#5481]
**Describe the bug**

Having installed kitty and bash through brew, starting kitty produces the following errors:

```
GNU bash, version 5.1.16(1)-release (aarch64-apple-darwin21.1.0)
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
-bash: /opt/homebrew/etc/bash_completion.d/brew: line 41: syntax error near unexpected token `<'
-bash: /opt/homebrew/etc/bash_completion.d/brew: line 41: `  while read -r line; do COMPREPLY+=("${line}"); done < <(compgen -W "${list}" -- "${cur}")'
-bash: /opt/homebrew/etc/bash_completion.d/git-completion.bash: line 3017: syntax error near unexpected token `<'
-bash: /opt/homebrew/etc/bash_completion.d/git-completion.bash: line 3017: `	done < <(git ls-tree -z -d --name-only HEAD $_tmp_dir)'
```

The same does not occur starting the standard macos terminal. I added a `bash --version` right before sourcing completion to make sure the PATH is correct and points to the recent version of bash.

**To Reproduce**
Steps to reproduce the behavior - very non-minimal instructions as I don't know if the exact environment is relevant or not:
1. Install bash and kitty from brew, endsure bash is in path (maybe not required)
2. Setup profile to initialize bash completion, with the completion files in this gist: https://gist.github.com/imsodin/5e2e21156dc807e6dd82a793c96d199a
3. Start kitty observe error
**Environment details**
```
macos 12.5, M1
```
Neither they shortcuts nor --debug-config worked for me (unknown option).

**Comment by kovidgoyal** [comment]
Don't install things through brew, lord knows what brew does to
upstream software. Use the official kitty binaries. If you can reproduce
with that, let me know and I will take a look.

**Comment by imsodin** [comment]
Yeah, sorry about the noise - brew messed it up, kitty dmg works. Thanks for the pointer.

/rant_start Me linux guy, I like command line package managers instead of downloading a weird format, drag and dropping it somewhere to install (what do I do on upgrade, same dance again!?) - whoever invented that *** (obviously noone at kitty) should get some anti-award. Also am a go application maintainer, so very much understand why only reproducers with official binaries count, super annoying having to deal with 3rd parties doing unexpected things - again very sorry I forgot about that, macos is already making me into a "try out random stuff until it works or you give up" kinda user.

**Comment by imsodin** [comment]
Edited the above comment, but just in case you already read that before the correction:

I can reproduce with .dmg from kitty releases.

**Comment by kovidgoyal** [comment]
You are almost certainly using the system bash (which is an antiquated piece of junk because Apple is scared of the GPL) rather than the brew bash. Add the following to kitty.conf and you should be fine

shell /opt/homebrew/bin/bash


Or for a more robust fix change the shell for your user account to the brew bash, then kityt will pick it up automatically.

**Comment by imsodin** [comment]
That was indeed it - I thought I did it, because another terminal worked and I set up PATH, but naturally kitty needs to know the shell before creating the shell. No idea why it works for others.

Anyway thanks a lot for your time and this great project!
